client/main.lua

```diff
@@ -90,145 +72,113 @@ local function RemoveTruckerBlips()
-        boxName = Config.Locations[type].label
-        size = 3
-    elseif type == "vehicle" then
-        event = "qb-truckerjob:client:Vehicle"
-        label = "Vehicle"
-        coords = vector3(Config.Locations[type].coords.x, Config.Locations[type].coords.y, Config.Locations[type].coords.z)
-        heading = Config.Locations[type].coords.h
-        boxName = Config.Locations[type].label
-        size = 5
-    elseif type == "stores" then
-        event = "qb-truckerjob:client:Store"
-        label = "Store"
-        coords = vector3(Config.Locations[type][number].coords.x, Config.Locations[type][number].coords.y, Config.Locations[type][number].coords.z)
-        heading = Config.Locations[type][number].coords.h
-        boxName = Config.Locations[type][number].name
-        size = 40
+    local rotation
+    local boxName
+    local icon
+    local debug
+
+    for k, v in pairs(Config.Locations) do
+        if k == type then
+            if type == 'stores' then
+                coords = v[number].coords
+                size = v[number].size
+                rotation = v[number].rotation
+                boxName = v[number].label
+                debug = v[number].debug
+            else
+                coords = v.coords
+                size = v.size
+                rotation = v.rotation
+                boxName = v.label
+                icon = v.icon
+                debug = v.debug
+            end
+        end
     end
-
-    if Config.UseTarget and type == "main" then
-        exports['qb-target']:AddBoxZone(boxName, coords, size, size, {
-            minZ = coords.z - 5.0,
-            maxZ = coords.z + 5.0,
-            name = boxName,
-            heading = heading,
-            debugPoly = false,
-        }, {
+    if Config.UseTarget and type == 'main' then
+        exports.ox_target:addBoxZone({
+            coords = coords,
+            size = size,
+            rotation = rotation,
+            debug = debug,
             options = {
                 {
-                    type = "client",
-                    event = event,
-                    label = label,
-                },
-            },
-            distance = 2
+                    name = boxName,
+                    event = 'qb-truckerjob:client:PaySlip',
+                    icon = icon,
+                    label = boxName,
+                    distance = 2,
+                }
+            }
         })
     else
-        local zone = BoxZone:Create(
-            coords, size, size, {
-                minZ = coords.z - 5.0,
-                maxZ = coords.z + 5.0,
-                name = boxName,
-                debugPoly = false,
-                heading = heading,
-            })
-
-        zoneCombo = ComboZone:Create({zone}, {name = boxName, debugPoly = false})
-        zoneCombo:onPlayerInOut(function(isPointInside)
-            if isPointInside then
-                if type == "main" then
+        CreateThread(function()
```

Comment:
Curious what the thoughts or requirements are to put this in a thread?

Reply:
Fixing this now